src/test/kotlin/uk/co/ben_gibson/git/link/url/GerritTest.kt

```diff
@@ -0,0 +1,73 @@
+package uk.co.ben_gibson.git.link.url
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import uk.co.ben_gibson.git.link.git.*
+import uk.co.ben_gibson.git.link.ui.LineSelection
+import java.util.stream.Stream
+import uk.co.ben_gibson.git.link.url.factory.TemplatedUrlFactory
+import uk.co.ben_gibson.git.link.url.template.UrlTemplates
+import uk.co.ben_gibson.url.URL
+
+class GerritTest {
+
+    companion object {
+
+        private val REMOTE_BASE_URL = URL.fromString("https://stash.example.com/foo/bar")
+        private const val BRANCH = "master"
+        private val COMMIT = Commit("b032a0707beac9a2f24b1b7d97ee4f7156de182c")
+        private val FILE = File("Foo.java", false, "src", false)
+        private val LINE_SELECTION = LineSelection(10, 20)
+
+        @JvmStatic
+        fun urlExpectationsProvider(): Stream<Arguments> = Stream.of(
+            Arguments.of(
+                UrlOptionsFileAtBranch(REMOTE_BASE_URL, FILE, BRANCH, LINE_SELECTION),
+                "https://stash.example.com/plugins/gitiles/foo/bar/+/refs/heads/master/src/Foo.java#10"
```

Comment:
Maybe put `https://stash.example.com/plugins/gitiles/foo/bar/+` in a `const val`?